multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/signature/SignatureKeyManager.java

```diff
@@ -46,7 +52,7 @@
  */
 @Beta
 @Singleton
-public class SignatureKeyManager {
+public class SignatureKeyManager implements EventSubscriber<WorkspaceStatusEvent> {
```

Comment:
I think implementing `EventSubscriber` be the manager is not a good idea. It would be better to use `EventSubscriber`  internally as it was before because it is not really public API of `SignatureKeyManager` class. Also, what if `SignatureKeyManager` will require one more subscription like `RuntimeStatusEvent`? It is not possible to implement the same interface with different generic types